test/models/content_view_test.rb

```diff
@@ -607,5 +607,17 @@ def test_import_only_immutable
       cv.import_only = false
       refute cv.valid?
     end
+
+    def test_update_host_statuses
+      other_location = taxonomies(:location2)
+      facet = katello_content_facets(:content_facet_two)
+
+      refute_equal facet.host.location, other_location
+      assert_nothing_raised do
+        Location.as_taxonomy(facet.host.organization, other_location) do
```

Comment:
Just curious, what does `as_taxonomy` do?

Reply:
The foreman application has 3 'states' that determine the scope of some query:
```
User.current
Location.current
Organization.current
```

(Note that these are Thread local variables, not class variables)  These are usually determined by what user is logged in, or the current location or organization that is set.  In this case, Location.as_taxonomy() sets the current Location and Organization temporarily within the block of code that is passed.  So even if Location.current is set to Location1 (or nil), it will override that and temporarily set Location.current to what is passed in.